Pilot/MessageSender.py

```diff
@@ -0,0 +1,266 @@
+    Return:
+      function: or None if requiredKeys is None
+  """
+  if not requiredKeys:
+    return None
+
+  def areParamsCorrect(params):
+    """
+      Args:
+        params(dict):
+      Return:
+        bool:
+    """
+    if not params:
+      return False
+    if not all(k in params for k in requiredKeys):
+      return False
+    return True
+  return areParamsCorrect
+
+
+class RESTSender(MessageSender):
+  """ Message sender to a REST interface.
+  """
+
+  REQUIRED_KEYS = ['HostKey', 'HostCertififcate',
+                   'CACertificate', 'Url', 'LocalOutputFile']
+
+  def __init__(self, params):
+    """
+      Raises:
+        ValueError: If params are not correct.
+    """
+    self._areParamsCorrect = createParamChecker(self.REQUIRED_KEYS)
+    self.params = params
+    if not self._areParamsCorrect(self.params):
+      raise ValueError("Parameters missing needed to send messages")
+
+  def sendMessage(self, msg, flag):
+    url = self.params.get('Url')
+    hostKey = self.params.get('HostKey')
+    hostCertificate = self.params.get('HostCertificate')
+    CACertificate = self.params.get('CACertificate')
+
+    try:
+      requests.post(url, json=msg, cert=(
+        hostCertificate, hostKey), verify=CACertificate)
+    except (requests.exceptions.RequestException, IOError) as e:
+      logging.error(e)
+      return False
+    return True
+
+
+def eraseFileContent(filename):
+  """ Erases the content of a given file.
+  """
+  with open(filename, 'w+') as myFile:
+    myFile.truncate()
+
+
+def saveMessageToFile(msg, filename='myLocalQueueOfMessages'):
+  """ Adds the message to a file appended as a next line.
+  """
+  with open(filename, 'a+') as myFile:
+    myFile.write(msg+'\n')
+
+
+def readMessagesFromFileAndEraseFileContent(filename='myLocalQueueOfMessages'):
+  """ Generates the queue FIFO and fills it
+      with values from the file, assuming that one line
+      corresponds to one message.
+      Finallym the file content is erased.
+  Returns:
+    Queue:
+  """
+  queue = Queue.Queue()
+  with open(filename, 'r') as myFile:
+    for line in myFile:
+      queue.put(line)
+  eraseFileContent(filename)
+  return queue
+
+
+class LocalFileSender(MessageSender):
+  """ Message sender to a local file.
+  """
+
+  REQUIRED_KEYS = ['LocalOutputFile']
+
+  def __init__(self, params):
+    """
+      Raises:
+        ValueError: If params are not correct.
+    """
+    self._areParamsCorrect = createParamChecker(self.REQUIRED_KEYS)
+    self.params = params
+    if not self._areParamsCorrect(self.params):
+      raise ValueError("Parameters missing needed to send messages")
+
+  def sendMessage(self, msg, flag):
+    filename = self.params.get('LocalOutputFile')
+    saveMessageToFile(msg, filename=filename)
+    return True
+
+
+def connect(hostAndPort, sslCfg):
+  """ Connects to RabbitMQ and returns connection
+      handler or None in case of connection down.
+      Stomp-depended function.
```

Comment:
What's the format expected for `hostAndPort` and `sslCfg` parameters?

Reply:
added